adr/ADR-001.md

```diff
@@ -0,0 +1,165 @@
+# ADR-01: New x/mint module.
+
+## Abstract
+
+We propose to replace the default cosmos-sdk `x/mint` module with an archway variation which inflates using block time
+instead of using block numbers. In order to make inflation more linear and predictable. 
+
+## Context
+
+The default cosmos-sdk `x/mint` module creates an inflation stream based on the foreseen amount of blocks that there can
+be in a single year. This is problematic, because the amounts of block in a year can change drastically based on the 
+network's decentralisation. For example, initially the chain might have `1 second` block-times, which means the foreseen
+amount of blocks in a single year is `31,536,000`, but as the network gets more decentralised block-times will grow due
+to network latency, maybe raising to `5` seconds. This causes the amount of blocks in a single year to change, and hence
+the amount of inflation happening each block should grow too, to follow the shift in block-times. This causes inflation to
+be lower than the expected one. There's another case for which if block-times become faster, then the network will inflate
+more than the expected.
+
+Unfortunately, in order to re-adjust block provision a governance proposal is required to change the `x/mint.Params.BlocksPerYear`
+parameter. Relying on governance to actively adjust the target inflation through the `BlocksPerYear` param is infeasible long
+term as there can be block-time shifts as the proposal is ongoing.
+
+Considering Archway Network relies on inflation to provide part of the inflationary rewards to developers, the rewards stream 
+needs to be predictable and stable.
```

Comment:
Was wondering with inflation changing based on amount of bonded tokens, wouldnt that make inflation even less predictable? 🤔

Reply:
yes. but we care about meeting a certain floor, the minimum inflation. Unfortunately cosmos-sdk's minting  cannot guarantee it due to the `BlocksPerYear`  parameter. Here we can guarantee a minimum, the invariance will never be broken unless there are some chain halts (hopefully not).